### PR TITLE
fix: keep label values valid after truncation

### PR DESCRIPTION
**Pull request by aarons-afk**
### What
`sanitize_label_value` maps invalid characters to `_` and caps the result at 63 characters, but a cut that lands on a `_`, `.` or `-` leaves a value K8s rejects with err: a label value must start and end with an alphanumeric.

This happens when the sanitized string is longer than 63 chars and character 63 is a separator. The plan ConfigMap is then rejected and the policy stops reconciling:

`metadata.labels: Invalid value:
"foo_params_literal_db.internal.example.com_literal_appd_":
a valid label must be an empty string or consist of alphanumeric characters,
'-', '_' or '.', and must start and end with an alphanumeric character
(regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')`

### Fix
Trim leading and trailing non-alphanumerics after truncation. `sanitize_label_value` is the single builder for every label value - used both when writing labels and in the selectors that read them back - so writes and lookups stay consistent, and only already-invalid values change. 

### Testing
- Adds a table test: unchanged, trimmed, empty, exactly-63 boundary, and truncated (char 63 a separator) cases.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved label value sanitization by removing leading and trailing separators.
  * Ensured labels remain valid when inputs contain only separators or when truncation ends with a separator.
* **Tests**
  * Added coverage for valid labels, separator handling, empty results, length limits, and truncation edge cases.

### CODE REVIEW

File: crates/pgroles-operator/src/plan.rs [hunk 1]
```diff
@@ -1121,7 +1120,10 @@ fn sanitize_label_value(value: &str) -> String {
         })
         .take(63)
         .collect();
+    // Truncation may land on a separator; trim so the value starts/ends alphanumeric.
     sanitized
+        .trim_matches(|c: char| !c.is_ascii_alphanumeric())
+        .to_string()
```
Comment (hardbyte) [inline crates/pgroles-operator/src/plan.rs:1132]: Non-blocking: trimming *after* `.take(63)` means leading separators consume the 63-char budget and are then discarded, so we lose characters from the meaningful tail (e.g. `"__" + "a"×70` yields 61 chars instead of 63). Trimming the start before truncation and only the end after keeps the full budget:

```suggestion
    let sanitized: String = value
        .trim_start_matches(|c: char| !c.is_ascii_alphanumeric())
        .chars()
        .map(|c| {
            if c.is_ascii_alphanumeric() || c == '.' || c == '-' || c == '_' {
                c
            } else {
                '_'
            }
        })
        .take(63)
        .collect();
    // Truncation may land on a separator; trim so the value ends alphanumeric.
    sanitized
        .trim_end_matches(|c: char| !c.is_ascii_alphanumeric())
        .to_string()
```

Not reachable through today's callers (both label values start with a namespace, so they always begin alphanumeric) — happy to take it as-is if you'd rather keep the diff minimal.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hardbyte — APPROVED** [review]
Thanks — good catch, and the fix is the right shape: `sanitize_label_value` is the single builder for both the written labels and the read-back selectors, so trimming here keeps writes and lookups consistent, and only values the API server was already rejecting change.

I verified this beyond the table test: fuzzed ~200k adversarial inputs (NULs, `=`, slashes, non-ASCII, separator-only, boundary lengths) through the new implementation against the K8s label regex — zero invalid outputs, all ≤63 bytes. The ASCII-only mapping also means chars == bytes, so `.take(63)` can't overshoot the byte limit. Tests pass locally.

One trade-off worth stating explicitly for the record: this converts a loud failure into a quiet aliasing. A name like `"a"×62 + "-b"` previously produced an invalid label and the reconcile failed visibly; it now trims to the same label as a policy literally named `"a"×62`. Since `pgroles.io/policy` selectors drive plan lookup, approval, and cleanup deletion, a collision would cross-wire two policies silently. I'm accepting that trade — it fixes a real stuck-reconcile outage, and 63-char-prefix collisions already existed via truncation before this PR — but the durable fix is to make the selector identity injective (e.g. a `pgroles.io/policy-uid` label used for all selectors, keeping the name label as human-readable metadata). That would also close the pre-existing gap in `main.rs` where the plan→policy watch mapping compares the truncated label against the raw name and silently never matches for >63-char policy names. I'll track that separately; it's the same pattern we already adopted for `LABEL_PLAN` with the hash-based `plan_label_value`.

Two tiny non-blocking notes besides the inline suggestion:
- The updated doc comment says values must start and end alphanumeric; worth mentioning that the empty string is also valid (your `"___"` → `""` test case relies on it).
- Empty output would make the selector `pgroles.io/policy=` match "key present with empty value" rather than fail. Unreachable today since policy names and namespaces are DNS-1123 (always contain an alphanumeric), but if this helper ever grows a user-controlled caller a fallback value or debug assert would be cheap insurance.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by hardbyte** [comment]
Thanks for this @aarons-afk. Closing because the fix has landed

Your change is in main as of 34f2ced (#152)

It grew a bit on the way in. Chasing your bug turned up that the operator restated Kubernetes' identifier rules inline at each call site, and the restatements disagreed with each other — so rather than patch `sanitize_label_value` in place, the rules now live in one module ([crates/pgroles-operator/src/k8s_names.rs](https://github.com/hardbyte/pgroles/blob/claude/release-0.8.0/crates/pgroles-operator/src/k8s_names.rs)) and your fix is the body of `LabelValue::sanitize`. The old `sanitize_label_value` in `plan.rs` is now a one-line delegation to it.

Two refinements on top of your version:

- Leading separators are trimmed before the 63-byte cut rather than after, so they don't consume budget that real characters could use.
- The truncation respects UTF-8 character boundaries.

Both are now pinned by property tests ([tests/identifier_properties.rs](https://github.com/hardbyte/pgroles/blob/claude/release-0.8.0/tests/identifier_properties.rs)) that run every builder against a hostile alphabet.

Unifying the rules also exposed two more latent bugs in the same family: `generate_plan_name` left a trailing separator exposed by its cut, and `is_valid_secret_name` rejected names starting with a digit, which Kubernetes actually permits. 

Ships in 0.8.0 (#153) soon
